lib/galaxy/job_execution/output_collect.py

```diff
@@ -638,15 +638,16 @@ def default_exit_code_file(files_dir, id_tag):
 
 def collect_extra_files(object_store, dataset, job_working_directory):
     file_name = dataset.dataset.extra_files_path_name_from(object_store)
-    temp_file_path = os.path.join(job_working_directory, "working", file_name)
+    output_dir = "working" if os.path.exists(os.path.join(job_working_directory, "working", file_name)) else "outputs"
```

Comment:
I'd feel more comfortable about this if we were choosing between the two based on the job type in some way.